### PR TITLE
Use NetworkInterface get mac address when ifconfig/getmac fails

### PR DESCRIPTION
**Pull request by Flanker32**
Use NetworkInterface get mac address when `ifconfig`/`getmac` fails.

### CODE REVIEW

File: azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/telemetry/GetHashMac.java [hunk 1]
```diff
@@ -13,40 +13,39 @@
...
-    public static boolean isValidHashMacFormat(String hashMac) {
-        if (hashMac == null || hashMac.isEmpty()) {
-            return false;
-        }
-
-        final Pattern hashedMacPattern = Pattern.compile(HASHED_MAC_REGEX);
-        final Matcher matcher = hashedMacPattern.matcher(hashMac);
-        return matcher.matches();
-    }
+    public static final Pattern MAC_PATTERN = Pattern.compile(MAC_REGEX);
```
Comment (andxu) [inline azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/telemetry/GetHashMac.java]: can MAC_PATTERN  be private?

File: azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/telemetry/GetHashMac.java [hunk 2]
```diff
@@ -85,6 +88,29 @@ private static String getRawMac() {
         return ret;
     }
 
+    private static String getRawMacWithoutIfconfig() {
+        final List<String> macSet = new ArrayList<>();
+        try {
+            final Enumeration<NetworkInterface> interfaces = NetworkInterface.getNetworkInterfaces();
```
Comment (andxu) [inline azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/telemetry/GetHashMac.java]: log an event when ifconfig is not available(this code is called)

Comment (Flanker32) [reply azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/telemetry/GetHashMac.java]: Done

File: azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/telemetry/GetHashMac.java (hunk 1 above)
Comment (Flanker32) [reply azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/telemetry/GetHashMac.java]: Yes, they are used just here.

File: azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/telemetry/GetHashMac.java (hunk 2 above)
Comment (Flanker32) [reply azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/telemetry/GetHashMac.java]: This method is called while Telemetry Proxy is initializing, so we can't send telemetries here.

**Pull request merged**